examples/hrtim/adc-trigger.rs

```diff
@@ -0,0 +1,150 @@
+#![no_std]
+#![no_main]
+
+use cortex_m_rt::entry;
+
+use defmt_rtt as _; // global logger
+use panic_probe as _;
+
+#[cfg(not(any(feature = "stm32g474", feature = "stm32g484")))]
+#[entry]
+fn main() -> ! {
+    #[allow(clippy::empty_loop)]
+    loop {}
+}
+
+#[cfg(any(feature = "stm32g474", feature = "stm32g484"))]
+#[entry]
+fn main() -> ! {
+    use hal::adc::{self, config::ExternalTrigger12};
+    use stm32g4xx_hal as hal;
+
+    use defmt::info;
+    use hal::{
+        adc::{
+            config::{Continuous, Dma as AdcDma, SampleTime, Sequence},
+            AdcClaim, ClockSource, Temperature, Vref,
+        },
+        delay::SYSTDelayExt,
+        dma::{self, config::DmaConfig, stream::DMAExt, TransferExt},
+        gpio::{gpioa::PA8, gpioa::PA9, Alternate, GpioExt, AF13},
+        hrtim::control::HrControltExt,
+        hrtim::output::HrOutput,
+        hrtim::HrPwmAdvExt,
+        hrtim::{control::Adc13Trigger, Pscl4},
+        pwr::PwrExt,
+        rcc::{self, RccExt},
+        stm32::Peripherals,
+    };
+
+    info!("start");
+
+    let dp = Peripherals::take().unwrap();
+    let cp = cortex_m::Peripherals::take().expect("cannot take core peripherals");
+
+    // Set system frequency to 16MHz * 15/1/2 = 120MHz
+    // This would lead to HrTim running at 120MHz * 32 = 3.84...
+    info!("rcc");
+    let pwr = dp.PWR.constrain().freeze();
+    let mut rcc = dp.RCC.freeze(
+        rcc::Config::pll().pll_cfg(rcc::PllConfig {
+            mux: rcc::PLLSrc::HSI,
+            n: rcc::PllNMul::MUL_15,
+            m: rcc::PllMDiv::DIV_1,
+            r: Some(rcc::PllRDiv::DIV_2),
+
+            ..Default::default()
+        }),
+        pwr,
+    );
+
+    let mut delay = cp.SYST.delay(&rcc.clocks);
+
+    let dma::stream::StreamsTuple(dma1ch1, ..) = dp.DMA1.split(&rcc);
+    let config = DmaConfig::default()
+        .transfer_complete_interrupt(true)
+        .circular_buffer(true)
+        .memory_increment(true);
+
+    info!("Setup Gpio");
+    let gpioa = dp.GPIOA.split(&mut rcc);
+    let pa0 = gpioa.pa0.into_analog();
+
+    info!("Setup Adc1");
+    let mut adc = dp
+        .ADC1
+        .claim(ClockSource::SystemClock, &rcc, &mut delay, true);
+
+    adc.set_external_trigger((
+        adc::config::TriggerMode::RisingEdge,
+        ExternalTrigger12::Hrtim_adc_trg_1,
```

Comment:
Here is yet another enum... What if this argument was a struct that could only be obtained by calling `hr_control.enable_adc_trigger1_source` or something